src/simple-scrollbar.js

```diff
@@ -68,31 +97,44 @@
 
     dragDealer(this.bar, this);
     this.moveBar();
-    
-    var moveBar = this.moveBar.bind(this)
-    
+    var moveBar = this.moveBar.bind(this);
+    const moveBarDebounced = debounce(200, moveBar);
+
     this.el.addEventListener('scroll', moveBar);
     this.el.addEventListener('mouseenter', moveBar);
 
-    this.target.classList.add('ss-container'); 
-      
+    this.target.classList.add('ss-container');
+
     var css = window.getComputedStyle(el);
-  	if (css['height'] === '0px' && css['max-height'] !== '0px') {
-    	el.style.height = css['max-height'];
+    if (css['height'] === '0px' && css['max-height'] !== '0px') {
+      el.style.height = css['max-height'];
     }
-
-    var mutationObserver = new MutationObserver(function(mutations){
+    if (el.offsetHeight === 0) {
+      var visibilityObserver = new MutationObserver(function (visibilityMutations) {
         moveBar();
+        visibilityObserver.disconnect();
+      });
+      var config = { attributes: true, childList: false, characterData: false, subtree: false, attributeFilter: ["class", "style"] };
+      var hiddenParent = el;
+      if (hiddenParent.style.display !== "none") {
```

Comment:
checking style.display is incorrect, it fails if it has a css class that sets display:none.
check offsetParent !== null instead